### PR TITLE
add EKS cluster terraform script

### PR DESCRIPTION
**Pull request by carlossscastro**
add EKS cluster terraform script

### CODE REVIEW

File: terraform_modules/base_framework/ekscluster.tf [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# EKS Cluster
+resource "aws_eks_cluster" "ekscluster" {
+  name     = "${var.canary_name}-Cluster"
+  role_arn = aws_iam_role.cluster.arn
+  version  = var.k8s_version
+
+  vpc_config {
+    subnet_ids              = flatten([aws_subnet.public_subnet[*].id, aws_subnet.private_subnets[*].id])
```
Comment (kang-makes) [inline terraform_modules/base_framework/ekscluster.tf:8]: Do we need to give EKS access to public subnets? That looks risky.

File: terraform_modules/base_framework/ekscluster.tf [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  value = aws_eks_cluster.ekscluster.name
+}
+
+output "cluster_endpoint" {
+  value = aws_eks_cluster.ekscluster.endpoint
+}
+
+output "cluster_ca_certificate" {
+  value = aws_eks_cluster.ekscluster.certificate_authority[0].data
+}
```
Comment (kang-makes) [inline terraform_modules/base_framework/ekscluster.tf:189]: missing EOL

File: terraform_modules/base_framework/vars.tf [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
+  default     = "1.24"
+}
+variable nodes_instance_type {
+  type    = string
+  default = "t2.medium" # https://aws.amazon.com/ec2/instance-types/
+}
+variable nodes_ami_type {
+  type    = string
+  default = "AL2_x86_64" # AL2_x86_64, AL2_x86_64_GPU, AL2_ARM_64, CUSTOM
+}
```
Comment (kang-makes) [inline terraform_modules/base_framework/vars.tf:33]: Missing EOL

**Review by kang-makes — COMMENTED** [review]
I am not comfortable with the EKS having access to the public subnet.

Did we test that the EKS cluster not only deploys but has internet access? Did we test that a pod is able to ping the bastion?

File: terraform_modules/base_framework/ekscluster.tf [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.eks_nodes.id
+  source_security_group_id = aws_security_group.eks_cluster.id
+  to_port                  = 65535
+  type                     = "ingress"
+}
+
+output "cluster_name" {
+  value = aws_eks_cluster.ekscluster.name
+}
```
[the remark this answers is not in the document]
Comment (carlossscastro) [reply terraform_modules/base_framework/ekscluster.tf]: Thanks. Changed to follow the same pattern

**Review by kang-makes — APPROVED** [review]
Awesome! LGTM :D

**Pull request merged**